src/main/java/org/owasp/benchmark/testcode/BenchmarkTest01101.java

```diff
@@ -0,0 +1,122 @@
+		response.setContentType("text/html;charset=UTF-8");
+	
+		String param = "";
+		java.util.Enumeration<String> names = request.getHeaderNames();
+		while (names.hasMoreElements()) {
+			String name = (String) names.nextElement();
+			
+			if(org.owasp.benchmark.helpers.Utils.commonHeaders.contains(name)){
+				continue;
+			}
+			
+			java.util.Enumeration<String> values = request.getHeaders(name);
+			if (values != null && values.hasMoreElements()) {
+				param = name;
+				break;
+			}
+		}
+		// Note: We don't URL decode header names because people don't normally do that
+
+		String bar = new Test().doSomething(request, param);
+		
+		// Code based on example from:
+		// http://examples.javacodegeeks.com/core-java/crypto/encrypt-decrypt-file-stream-with-des/
+
+		try {
+			javax.crypto.Cipher c = org.owasp.benchmark.helpers.Utils.getCipher();
+			// encrypt and store the results
+			byte[] input = { (byte)'?' };
+			Object inputParam = bar;
+			if (inputParam instanceof String) input = ((String) inputParam).getBytes();
+			if (inputParam instanceof java.io.InputStream) {
+				byte[] strInput = new byte[1000];
+				int i = ((java.io.InputStream) inputParam).read(strInput);
+				if (i == -1) {
+					response.getWriter().println(
+"This input source requires a POST, not a GET. Incompatible UI for the InputStream source."
+);
+					return;
+				}
+				input = java.util.Arrays.copyOf(strInput, i);
+			}
+			byte[] result = c.doFinal(input);
+			
+			java.io.File fileTarget = new java.io.File(
+					new java.io.File(org.owasp.benchmark.helpers.Utils.testfileDir),"passwordFile.txt");
+			java.io.FileWriter fw = new java.io.FileWriter(fileTarget,true); //the true will append the new data
+			    fw.write("secret_value=" + org.owasp.esapi.ESAPI.encoder().encodeForBase64(result, true) + "\n");
+			fw.close();
+			response.getWriter().println(
+"Sensitive value: '" + org.owasp.esapi.ESAPI.encoder().encodeForHTML(new String(input)) + "' encrypted and stored<br/>"
+);
+
+		} catch (javax.crypto.IllegalBlockSizeException e) {
+			response.getWriter().println(
+"Problem executing crypto - javax.crypto.Cipher.getInstance(java.lang.String,java.security.Provider) Test Case"
+);
+e.printStackTrace(response.getWriter());
+			throw new ServletException(e);
+		} catch (javax.crypto.BadPaddingException e) {
+			response.getWriter().println(
+"Problem executing crypto - javax.crypto.Cipher.getInstance(java.lang.String,java.security.Provider) Test Case"
+);
+e.printStackTrace(response.getWriter());
+			throw new ServletException(e);
+		}
+		response.getWriter().println(
+"Crypto Test javax.crypto.Cipher.getInstance(java.lang.String,java.lang.String) executed"
+);
+	}  // end doPost
+
+	
+    private class Test {
+
+        public String doSomething(HttpServletRequest request, String param) throws ServletException, IOException {
+
+		String bar = org.apache.commons.lang.StringEscapeUtils.escapeHtml(param);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

escapeJava(), escapeHtml3(), escapeHtml4(), escapeXml(), escapeXml10(), and escapeXml11() have been deprecated in org.apache.commons.lang.StringEscapeUtils and org.apache.commons.lang3.StringEscapeUtils and do not cover the entire range of characters that have to be encoded. OWASP Java Encoder library should be used instead.